Modules/xASL_module_ASL.m

```diff
@@ -407,14 +407,16 @@
 x.P.Path_PWI4D_used = x.P.Path_PWI4D;
 x.P.Pop_Path_PWI4D_used = x.P.Pop_Path_PWI4D;
 
-if x.modules.asl.bMergingSessions
-    % only if we merged sessions, we use custom paths
-	[x.P.Path_PWI4D_used, x.P.Pop_Path_PWI4D_used] = xASL_im_MergePWI4D(x);
-	% Note: The paths x.P.(Pop_)Path_PWI4D_used can differ from a PWI4D obtained from a simple subtraction as we may have merged sessions, 
-	% and the user or ExploreASL may have removed volumes from PWI4D. That's why we save and use x.P.(Pop_)Path_PWI4D_used in memory, such 
-	% that the correct NIfTI will be used later in the pipeline (e.g., in xASL_wrp_VisualQC)
+if (x.mutex.HasState(StateName{iState-4}) && (~x.mutex.HasState(StateName{iState}))) ||...
```

Comment:
Simplification:

You can remove (x.mutex.HasState(StateName{iState-4}) because 1) without resampling done, it wouldn't get here (it doesn't proceed to mutex+1 if mutex hasn't been completed), so we could write this for each mutex. Plus, this 2) becomes buggy if we would change the order of mutexes. So I would keep this simple. Plus 3) if this were to be the case, we actually want a crash! We rather want a crash than wrong results... (consider that a user could get results without taking the merged PWI4D but unmerged one)

We hardly use WADQC, so we could also move WADQC as step 5 of visualQC to make this code easier; we don't have any other dependencies of the mutex stats being n=10 right?

If you agree I can do this last step. Because now it is a bit unreadable when the merging is performed or not, we need to keep things reasonably simple (i.e., 1% exceptions like WADQC and 5% exceptions like merging shouldn't make the code too much convolved)